ui/lib/sync/addon/components/secrets/page/overview.hbs

```diff
@@ -2,28 +2,37 @@
   Copyright (c) HashiCorp, Inc.
   SPDX-License-Identifier: BUSL-1.1
 ~}}
-
-{{#if (and this.version.hasSecretsSync (not @isActivated))}}
-  {{#unless this.hideOptIn}}
-    <Hds::Alert
-      @type="inline"
-      @color="warning"
-      @onDismiss={{fn (mut this.hideOptIn) true}}
-      data-test-secrets-sync-opt-in-banner
-      as |A|
-    >
-      <A.Title>Enable Secrets Sync feature</A.Title>
-      <A.Description>To use this feature, specific activation is required. Please review the feature documentation and enable
-        it. If you're upgrading from beta, your previous data will be accessible after activation.</A.Description>
-      <A.Button
-        @text="Enable"
-        @color="secondary"
-        {{on "click" (fn (mut this.showActivateSecretsSyncModal) true)}}
-        data-test-secrets-sync-opt-in-banner-enable
-      />
-    </Hds::Alert>
-  {{/unless}}
-{{/if}}
+{{#unless @isActivated}}
+  {{#if (or @licenseHasSecretsSync @isHvdManaged)}}
+    {{#unless this.hideOptIn}}
```

Comment:
These if statements 😵 I wonder if we can move this logic to a backing class?

Reply:
Let me add this as a todo and revisit when Noelle returns. There's been a lot of back and forth on how to work through this logic (a persona service, etc). Additionally, there's a solid chance that an api change will clean some of this logic up.